tools/osv-scanner/main.go

```diff
@@ -37,12 +37,15 @@ func scanDir(query *osv.BatchedQuery, dir string) error {
 
 		if !info.IsDir() {
 			if parser, _ := lockfile.FindParser(path, ""); parser != nil {
-				scanLockfile(query, path)
+				err := scanLockfile(query, path)
 				if err != nil {
 					log.Println("Attempted to scan lockfile but failed: " + path)
 				}
 			}
-			scanSBOMFile(query, path)
+			extension := strings.ToLower(filepath.Ext(info.Name()))
+			if extension == ".json" || strings.Contains(info.Name(), ".spdx") {
```

Comment:
I'm not understanding why this can't be `extension == ".spdx"`

Could you either change it to check the same way as the `.json` or add a comment explaining the need for this?

Reply:
Based on the comment in `scanSBOMFile` it sounds like checking for the `.json` extension is redundant?

Reply:
> Based on the comment in scanSBOMFile it sounds like checking for the .json extension is redundant?

Ah yes, I think you're right I can remove this check out here.

> I'm not understanding why this can't be extension == ".spdx"

According to this: https://spdx.github.io/spdx-spec/v2.3/conformance/
SPDX files can have multiple different formats:

Format | Extension
-- | --
tag:value | *.spdx
RDF | *.spdx.rdf
JSON | *.spdx.json
XML | *.spdx.xml
YAML | *.spdx.yaml or *.spdx.yml

I'll add a comment on the check with the link.

Reply:
Ah, thanks for the explanation.